docs/proposals/image-updater-first-class-support.md

```diff
@@ -0,0 +1,310 @@
+Aside from providing strict type validation, converting these fields to first class fields in the CR would simplify the expression and usage of these fields and make them more intuitive for users to understand and leverage. 
+
+Example of proposed additions to Application CR spec and status to accomodate image update configuration is provided below:
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  ...
+
+spec:
+...
+
+  image:
+    # updates contains all the fields to configure exactly which images should be considered for automatic updates and how they should be updated
+    updates:
+      # ImageList contains list of image specific configuration for each image that is to be considered as a candidate for updation 
+      imageList:
+        - image: quay.io/some/image 
+
+          # (optional) Specify Semver constraint for allowed tags 
+          constraint: 1.17.x
+
+          # (optional) specify list of regex/wildcard patterns to further filter which tags should be considered for updates
+          allowTags: 
+            matchType: regex/wildcard
+            matchList:
+            - <pattern 1>
+            - <pattern 2>
+
+          # (optional) specify list of regex/wildcard patterns to dictate which tags should be ignored for updates
+          ignoreTags: 
+            matchType: regex/wildcard
+            matchList:
+            - <pattern 1>
+            - <pattern 2>
+
+          # (optional) force updates to images that don't appear in the status of an Argo CD application. Default is false   
+          forceUpdate: true/false
+
+          # specify strategy to be followed when determining which images have qualifying updates 
+          updateStrategy: semver/digest/lexical/most-recently-built
+
+          # specify list of architectures to be considered for a specific image if application is to be run across multiple clusters of differing architectures
+          imagePlatforms:
+          - linux/amd64
+          - linux/arm64
+
+          # (optional) pull secret for image 
+          pullSecret: 
+            namespace: argocd
+            name: image-updater-pull-secret
+            field: pull-secret-creds
+            
+            # alternatively, if using an env var instead of secrets
+            env: DOCKER_HUB_CREDDS
+
+            # or use an external script mounted to image-updater FS to generate credentials
+            ext: <path/to/script>
+
+          # (optional) specify helm parameter names to be used for image update write back
+          helm:
+            imageName: someImage.image  
```

Comment:
I know this is the existing config, but in the future can we consider not requiring `helm.imageName` if a user is updating only one image?

Reply:
The problem here is that we don't know (or can make an informed guess on) the name of the parameter to set. There is a default that we use, that probably works with a couple of off-the-shelve charts.